### ISSUE TITLE
Returning from operator[]= is void context.

### ISSUE THREAD

**Issue by peter-ahe-google** [dart-lang/kernel#31]
I've noticed that code like this:

```dart
class Context {
  final List list = [];
  operator[]= (int i, value) => list[i] = value;
}
```

Generates a lot of let expressions in the method body of operator[]=. I think this may be a common pattern, and that this is a void context, so it may be worth tweaking this.

**Comment by asgerf** [comment]
I sent you a CL with a fix.

**Comment by asgerf** [comment]
Fixed in 239d77596348966f7e3523452ab0caa84256bf62.
